### ISSUE TITLE
Crash during code completion inside of lambda functions

### ISSUE THREAD

**Issue by JohnDoneth** [godotengine/godot#50673]
### Godot version

Godot Engine v4.0.dev.custom_build.e248d2629

### System information

Arch Linux

### Issue description

When code completion inside of lambdas is invoked, the editor will crash.

```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x3cda0) [0x7f8e66ddfda0] (??:0)
[2] StringName::hash() const (/home/john/projects/godot/./core/string/string_name.h:112)
[3] HashMapHasherDefault::hash(StringName const&) (/home/john/projects/godot/./core/templates/hashfuncs.h:171)
[4] HashMap<StringName, MethodBind*, HashMapHasherDefault, HashMapComparatorDefault<StringName>, (unsigned char)3, (unsigned char)8>::get_element(StringName const&) const (/home/john/projects/godot/./core/templates/hash_map.h:175)
[5] HashMap<StringName, MethodBind*, HashMapHasherDefault, HashMapComparatorDefault<StringName>, (unsigned char)3, (unsigned char)8>::getptr(StringName const&) (/home/john/projects/godot/./core/templates/hash_map.h:303)
[6] ClassDB::get_method_info(StringName const&, StringName const&, MethodInfo*, bool, bool) (/home/john/projects/godot/core/object/class_db.cpp:674)
[7] /home/john/projects/godot/bin/godot.linuxbsd.tools.64() [0x2f6191c] (/home/john/projects/godot/modules/gdscript/gdscript_editor.cpp:1770)
[8] /home/john/projects/godot/bin/godot.linuxbsd.tools.64() [0x2f5df1b] (/home/john/projects/godot/modules/gdscript/gdscript_editor.cpp:1196)
[9] GDScriptLanguage::complete_code(String const&, String const&, Object*, List<ScriptCodeCompletionOption, DefaultAllocator>*, bool&, String&) (/home/john/projects/godot/modules/gdscript/gdscript_editor.cpp:2517)
[10] ScriptTextEditor::_code_complete_script(String const&, List<ScriptCodeCompletionOption, DefaultAllocator>*, bool&) (/home/john/projects/godot/editor/plugins/script_text_editor.cpp:686 (discriminator 2))
[11] ScriptTextEditor::_code_complete_scripts(void*, String const&, List<ScriptCodeCompletionOption, DefaultAllocator>*, bool&) (/home/john/projects/godot/editor/plugins/script_text_editor.cpp:675)
[12] CodeTextEditor::_complete_request() (/home/john/projects/godot/editor/code_editor.cpp:853)
[13] void call_with_variant_args_helper<CodeTextEditor>(CodeTextEditor*, void (CodeTextEditor::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (/home/john/projects/godot/./core/variant/binder_common.h:223 (discriminator 4))
[14] void call_with_variant_args<CodeTextEditor>(CodeTextEditor*, void (CodeTextEditor::*)(), Variant const**, int, Callable::CallError&) (/home/john/projects/godot/./core/variant/binder_common.h:333)
[15] CallableCustomMethodPointer<CodeTextEditor>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/john/projects/godot/./core/object/callable_method_pointer.h:97)
[16] Callable::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/john/projects/godot/core/variant/callable.cpp:64)
[17] Object::emit_signal(StringName const&, Variant const**, int) (/home/john/projects/godot/core/object/object.cpp:1100)
[18] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/john/projects/godot/core/object/object.cpp:1155)
[19] CodeEdit::request_code_completion(bool) (/home/john/projects/godot/scene/gui/code_edit.cpp:1562 (discriminator 2))
[20] CodeTextEditor::_code_complete_timer_timeout() (/home/john/projects/godot/editor/code_editor.cpp:842)
[21] void call_with_variant_args_helper<CodeTextEditor>(CodeTextEditor*, void (CodeTextEditor::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (/home/john/projects/godot/./core/variant/binder_common.h:223 (discriminator 4))
[22] void call_with_variant_args<CodeTextEditor>(CodeTextEditor*, void (CodeTextEditor::*)(), Variant const**, int, Callable::CallError&) (/home/john/projects/godot/./core/variant/binder_common.h:333)
[23] CallableCustomMethodPointer<CodeTextEditor>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/john/projects/godot/./core/object/callable_method_pointer.h:97)
[24] Callable::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/john/projects/godot/core/variant/callable.cpp:64)
[25] Object::emit_signal(StringName const&, Variant const**, int) (/home/john/projects/godot/core/object/object.cpp:1100)
[26] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/john/projects/godot/core/object/object.cpp:1155)
[27] Timer::_notification(int) (/home/john/projects/godot/scene/main/timer.cpp:61 (discriminator 2))
[28] Timer::_notificationv(int, bool) (/home/john/projects/godot/scene/main/timer.h:37 (discriminator 14))
[29] Object::notification(int, bool) (/home/john/projects/godot/core/object/object.cpp:843)
[30] SceneTree::_notify_group_pause(StringName const&, int) (/home/john/projects/godot/scene/main/scene_tree.cpp:839)
[31] SceneTree::process(float) (/home/john/projects/godot/scene/main/scene_tree.cpp:446 (discriminator 2))
[32] Main::iteration() (/home/john/projects/godot/main/main.cpp:2522)
[33] OS_LinuxBSD::run() (/home/john/projects/godot/platform/linuxbsd/os_linuxbsd.cpp:278)
[34] /home/john/projects/godot/bin/godot.linuxbsd.tools.64(main+0x16e) [0x20987a4] (/home/john/projects/godot/platform/linuxbsd/godot_linuxbsd.cpp:60)
[35] /usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7f8e66dcab25] (??:0)
[36] /home/john/projects/godot/bin/godot.linuxbsd.tools.64(_start+0x2e) [0x209857e] (??:?)
```

### Steps to reproduce

Enter the following code into a GDScript. I created a new project with a Node3D and attached a script to it to get to this minimal reproduction of the error:
```gdscript
extends Node3D

var lambda = func(x):
	x
#        ^--- Then type a period here, to invoke completion. It will crash.
```

**Comment by cdemirer** [comment]
Might be fixed, can't reproduce on 28174d531

**Comment by Calinou** [comment]
I can't reproduce this on 28174d531 either, closing.
